python/nano/src/bigdl/nano/pytorch/amp/bfloat16.py

```diff
@@ -0,0 +1,108 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import contextlib
+import io
+from logging import info, warning
+
+from pytorch_lightning import LightningModule
+import torch
+import subprocess
+import os
+from bigdl.nano.utils.log4Error import invalidOperationError
+
+
+class autocast(torch.cpu.amp.autocast):
+    r"""
+    Customized autocast to verify BF16 support.
+    """
+    def __init__(self):
+        super().__init__()
+
+    def __enter__(self):
+        self.global_max_cpu_isa = os.environ["ONEDNN_MAX_CPU_ISA"]
+        os.environ["ONEDNN_MAX_CPU_ISA"] = "ALL"
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        os.environ["ONEDNN_MAX_CPU_ISA"] = self.global_max_cpu_isa
+        return super().__exit__(exc_type, exc_val, exc_tb)
+
+
+class BF16Model(LightningModule):
+    def __init__(self, model, force_bf16=True):
+        super().__init__()
+        self._force_bf16 = force_bf16
+        self.bf16_model = model.bfloat16()
+
+    @property
+    def has_bf16_isa(self):
+        msg = subprocess.check_output("lscpu | grep bf16")
+        return len(msg) > 0
+
+    @torch.cpu.amp.autocast(dtype=torch.bfloat16)
+    def forward(self, *args, **kwargs):
+        _is_bf16 = getattr(self, "_is_bf16", None)
+        if _is_bf16 is None:
+            if self.has_bf16_isa:
+                os.environ['DNNL_VERBOSE'] = '1'
+                dnnl_log = io.StringIO()
+                with contextlib.redirect_stdout(dnnl_log):
+                    self.bf16_model(*args, **kwargs)
+                dnnl_log = dnnl_log.getvalue()
+                self._is_bf16 = True
+                _is_bf16 = True
+                if 'amx_bf16' in dnnl_log:
+                    info("AMX BF16 support is enabled in this run.")
+                elif 'avx512_core_bf16' in dnnl_log:
+                    info("AVX512 BF16 support is enabled in this run.")
+                else:
+                    if self._force_bf16:
```

Comment:
you can remove the if
invalidOperationError(!self._force_bf16,
                            errMsg="BF16 ISA support is not enabled under current context.",
                            fixMsg="Please try to upgrade your pytorch version to obtain"
                                   " BF16 acceleration."
                        )